test/parallel/test-http-outgoing-finish-writable.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const server = http.createServer(common.mustCall(function(req, res) {
+  assert(res.writable, 'Res should be writable when it is received \
+                          and opened.');
+  assert(!res.finished, 'Res shouldn\'t be finished when it is received \
+                          and opened.');
+  res.end();
+  assert(!res.writable, 'Res shouldn\'t be writable after it was closed.');
+  assert(res.finished, 'Res should be finished after it was closed.');
+
+  server.close();
+}));
+
+server.listen(0);
+
+server.on('listening', common.mustCall(function() {
+  http.request({
+    port: server.address().port,
+    method: 'GET',
+    path: '/'
+  }).end();
```

Comment:
What `http.request`  returns is an `OutgoingMessage` as well, we might want to change that it there as well.